src/mono/mono/mini/interp/transform.c

```diff
@@ -6034,19 +6034,10 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 					/* TODO: metadata-update: implement me. If it's an added field, emit a call to the helper method instead of MINT_LDFLDA_UNSAFE */
 					g_assert (!m_field_is_from_update (field));
 					int foffset = m_class_is_valuetype (klass) ? m_field_get_offset (field) - MONO_ABI_SIZEOF (MonoObject) : m_field_get_offset (field);
-					if (td->sp->type == STACK_TYPE_O) {
```

Comment:
I don't think `STACK_TYPE_MP` should be null. Wouldn't the following more conservative fix work instead:
```
if (td->sp->type == STACK_TYPE_O || td->sp->type == STACK_TYPE_I) {
```

Reply:
I need it to perform a null-check with `STACK_TYPE_MP` as LDFLDA must throw if it is given a null pointer. I don't think this change would achieve that.

Reply:
I checked locally the testcase from `src/tests/JIT/Regression/JitBlue/Runtime_77640/Runtime_77640.cs` and the type of the pointer is indeed `STACK_TYPE_I` not `STACK_TYPE_MP`. Did you run also some other test ?

Reply:
Your fix seem to do the trick, as well. Strange, I thought that pointers are `STACK_TYPE_MP`.

Reply:
Managed pointers can also be null.

Example:
```cs
Problem(ref *(StructWithField*)null);

static ref int Problem(ref StructWithField p) => ref p.Field;

struct StructWithField
{
    public int Field;
}
```
(A more real-world example would involved `default` ref structs with `ref` fields)